runtimevar/runtimevar.go

```diff
@@ -278,6 +280,23 @@ func (c *Variable) Latest(ctx context.Context) (Snapshot, error) {
 	return Snapshot{}, c.lastErr
 }
 
+// CheckHealth returns an error unless Latest will return a good value
+// without blocking.
+func (c *Variable) CheckHealth() error {
+	haveGood := false
+	select {
+	case <-c.haveGood:
+		haveGood = true
+	default:
+	}
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if haveGood && c.lastErr != ErrClosed {
+		return nil
+	}
+	return c.lastErr
```

Comment:
Is `c.lastErr` guaranteed to be non-nil? What if the RPC hasn't completed yet?

Reply:
`c.lastErr` is initialized to `errors.New("no value yet")` in the constructor, so it's always non-nil unless there's a good value.